### PR TITLE
Feat/enemy damage

### PR DESCRIPTION
**Pull request by NathanGr33n**
Enemies can damage and kill player now.

### CODE REVIEW

File: TKGame/BackEnd/EntityManager.cs [hunk 1]
```diff
@@ -147,5 +150,16 @@ public static void DamageEnemy()
...
             }
         }
+
+        public static void EnemyDamage()
+        {
+            foreach(Entity entity in entities)
+            {
+                if (entity != entities[0] && entity.hitBox.Intersects(entities[0].hitBox) && entity.isEnemy)
+                {
+                    entities[0].health -= 10;
```
Comment (NateCastleOIT) [inline TKGame/BackEnd/EntityManager.cs:160]: This is gonna happen once every frame so at least in my case 240 times in a second so won't this instantly kill the player?
We should probably limit the number of times the damage can be inflicted. See the TimeElapsed and related usage for the fire powerups for reference.

File: TKGame/Entity.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public abstract class Entity : ICollideComponent
         public Texture2D healthTexture;
         public bool needsHealth = false;
         public float originalHealth;
-        
+        public bool isEnemy = false;
```
Comment (NateCastleOIT) [inline TKGame/Entity.cs:53]: this doesn't seem very scalable for other entities/seems like bloat

File: TKGame/BackEnd/EntityManager.cs [hunk 3]
```diff
@@ -147,5 +150,16 @@ public static void DamageEnemy()
...
+                }
+            }
+        }
```
Comment (aaronw84) [inline TKGame/BackEnd/EntityManager.cs:163]: Just add comments here to specify.

**Review by aaronw84 — APPROVED** [review]
If it works, it's good.

**Review by NateCastleOIT — COMMENTED** [review]
This would insta kill the player, no?

File: TKGame/BackEnd/EntityManager.cs (hunk 1 above)
Comment (NathanGr33n) [reply TKGame/BackEnd/EntityManager.cs:160]: I fixed this with a simple counter int in then function and only decrementing  the health once a second.

File: TKGame/BackEnd/EntityManager.cs [hunk 4]
```diff
@@ -125,7 +128,7 @@ public static void DrawHealth(SpriteBatch spriteBatch, Entity entity)
             if (entity.needsHealth) //Makes sure that items don't get health bars
             {
                 entity.healthTexture = new Texture2D(spriteBatch.GraphicsDevice, 1, 1);
-                entity.healthBar = new Rectangle((int)entity.Position.X - 50, (int)entity.Position.Y - 70, 100, 10);
+                entity.healthBar = new Rectangle((int)entity.Position.X - 50, (int)entity.Position.Y - 70, (int)entity.health, 10);
```
Comment (NateCastleOIT) [inline TKGame/BackEnd/EntityManager.cs:131]: makes sense

File: TKGame/BackEnd/EntityManager.cs [hunk 5]
```diff
@@ -136,13 +139,44 @@ public static void DrawHealth(SpriteBatch spriteBatch, Entity entity)
         /// </summary>
         public static void DamageEnemy()
         {
-            foreach (Entity entity in entities)
+            int count = 0;
+            foreach(Entity entity in entities)
             {
-                if (entities[0].entityTexture == Art.PlayerTexture) //Checks if player has sword out
+                if(entity.isEnemy && entity.hitBox.Intersects(entities[0].weapon.hitbox))
```
Comment (NateCastleOIT) [inline TKGame/BackEnd/EntityManager.cs:145]: this logic makes a lot more sense

File: TKGame/BackEnd/EntityManager.cs [hunk 6]
```diff
@@ -136,13 +139,44 @@ public static void DrawHealth(SpriteBatch spriteBatch, Entity entity)
...
+                            entities[1].health -= entities[0].weapon.damageStat;
+                        }
+                        else if(entity == entities[2])
+                        {
+                            entities[2].health -= entities[0].weapon.damageStat;
+                        }
+                        else if(entity == entities[3])
+                        {
+                            entities[3].health -= entities[0].weapon.damageStat;
+                        }
```
Comment (NateCastleOIT) [inline TKGame/BackEnd/EntityManager.cs:160]: uh, why was this added? This looks like you can now only damage the first 3 entities that exist? Also, this is just a for loop transformed into a bunch of if elses. Confused about this change.

File: TKGame/BackEnd/EntityManager.cs [hunk 7]
```diff
@@ -136,13 +139,44 @@ public static void DrawHealth(SpriteBatch spriteBatch, Entity entity)
...
+        /// Checks if the player hitbox intersects with any enemy hitboxes and then decrements player health
+        /// </summary>
+        public static void EnemyDamage()
+        {
+            int count = 0;
+            foreach(Entity entity in entities)
+            {
+                if ( entity.hitBox.Intersects(entities[0].hitBox) && entity.isEnemy)
+                {
+                    if (count % 240 == 0)
```
Comment (NateCastleOIT) [inline TKGame/BackEnd/EntityManager.cs:177]: this is a very hacky method of doing this. We can't guarantee 240 frames per sec and I don't see count get incremented anywhere so 0 % 240 = 240 is always the case

File: TKGame/BackEnd/EntityManager.cs [hunk 8]
```diff
@@ -136,13 +139,44 @@ public static void DrawHealth(SpriteBatch spriteBatch, Entity entity)
...
         {
-            foreach (Entity entity in entities)
+            int count = 0;
+            foreach(Entity entity in entities)
             {
-                if (entities[0].entityTexture == Art.PlayerTexture) //Checks if player has sword out
+                if(entity.isEnemy && entity.hitBox.Intersects(entities[0].weapon.hitbox))
                 {
-                    if (entities[0].weapon.hitbox.Intersects(entity.hitBox) && entity != entities[0]) //Checks if hitboxes intersect and is not the player
+                    if (count % 240 == 0)
```
Comment (NateCastleOIT) [inline TKGame/BackEnd/EntityManager.cs:147]: hacky, count always = 0

**Review by NateCastleOIT — CHANGES_REQUESTED** [review]
I'm confused about some of these changes.

**Review by NateCastleOIT — APPROVED** [review]
still not enjoying the if else stuff but otherwise good

**Review by keeganhauser — APPROVED** [review]
Nothing additional to add. Seems like @NateCastleOIT got the main concerns picked out already
